### PR TITLE
Use custom commands instead of observers for spawning

### PR DESCRIPTION
**Pull request by janhohenheim**
Part of #203
Alternative to #231
Does *not* help with #223.

### CODE REVIEW

**Comment by janhohenheim** [comment]
@alice-i-cecile is this what you had in mind for the simple but limited spawning? Or would you prefer if I used a struct `SpawnPlayer` that the user then manually `add`s?

File: src/demo/level.rs [hunk 1]
```diff
@@ -1,18 +1,22 @@
 //! Spawn the main level by triggering other observers.
```
Comment (benfrankel) [inline src/demo/level.rs:1]: Outdated. Also, implementation details shouldn't be exposed in module-level docs.

File: src/demo/player.rs [hunk 2]
```diff
@@ -2,7 +2,10 @@
 //! Note that this is separate from the `movement` module as that could be used
 //! for other characters as well.
 
-use bevy::prelude::*;
+use bevy::{
+    ecs::{system::RunSystemOnce, world::Command},
```
Comment (benfrankel) [inline src/demo/player.rs:6]:
```suggestion
    ecs::{system::RunSystemOnce as _, world::Command},
```
Optional.

File: src/demo/player.rs [hunk 3]
```diff
@@ -25,15 +27,22 @@ pub(super) fn plugin(app: &mut App) {
...
 
-#[derive(Event, Debug)]
-pub struct SpawnPlayer;
-
 #[derive(Component, Debug, Clone, Copy, PartialEq, Eq, Default, Reflect)]
 #[reflect(Component)]
 pub struct Player;
 
+/// Command to spawn the player character.
+/// We can add fields to this struct if we need data for spawning.
```
Comment (benfrankel) [inline src/demo/player.rs:35]:
```suggestion
/// A command to spawn the player character.
```
I don't like the second line here personally. There are a lot of things the user _could_ do to expand on the template, and we're not going to list them all. At the very least this doesn't make sense as a doc comment.

File: src/demo/level.rs (hunk 1 above)
Comment (janhohenheim) [reply src/demo/level.rs]: [`e56ec7d` (#242)](https://github.com/TheBevyFlock/bevy_quickstart/pull/242/commits/e56ec7d77cf71e25957602d5abe7b94b441c1895)

File: src/demo/player.rs (hunk 3 above)
Comment (janhohenheim) [reply src/demo/player.rs]: We do however mention things that have a very high probability of coming up and are low-hanging expansions.
Moved the comment into the struct: [`d2c59b6` (#242)](https://github.com/TheBevyFlock/bevy_quickstart/pull/242/commits/d2c59b6ef9162e66971ade4188e99d4246ee75b0)

**Review by benfrankel — APPROVED** [review]
Approved. I think `spawn_level` should spawn a parent level entity with a `Level` marker component ideally, but this PR doesn't have to change the current behavior.

File: src/demo/level.rs [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
 
-fn spawn_level(_trigger: Trigger<SpawnLevel>, mut commands: Commands) {
-    // The only thing we have in our level is a player,
-    // but add things like walls etc. here.
-    commands.trigger(SpawnPlayer);
+impl Command for SpawnLevel {
+    fn apply(self, world: &mut World) {
+        // The only thing we have in our level is a player,
+        // but add things like walls etc. here.
+        world.commands().add(SpawnPlayer { max_speed: 400.0 });
```
Comment (benfrankel) [inline src/demo/level.rs:20]: Might need to do `world.flush_commands()` after this

**Review by benfrankel — APPROVED** [review]

File: src/demo/level.rs (hunk 4 above)
Comment (janhohenheim) [reply src/demo/level.rs:20]: [`41c6194` (#242)](https://github.com/TheBevyFlock/bevy_quickstart/pull/242/commits/41c6194bb969721cf5c4bd472c7e482c0fa7132c)

**Pull request merged**